python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -792,13 +790,15 @@ def df2RefCat(self, dfList, exposureBBox, exposureWcs):
         df = pd.concat(dfList)
         # translate ra/decl coords in dataframe to detector pixel coords
         # to down select rows that overlap the detector bbox
+        exposureWcs = exposure.getWcs()
+        exposureBBox = exposure.getBBox()
         mapping = exposureWcs.getTransform().getMapping()
         x, y = mapping.applyInverse(np.array(df[['ra', 'decl']].values*2*np.pi/360).T)
         inBBox = exposureBBox.contains(x, y)
-        refCat = self.df2SourceCat(df[inBBox])
+        refCat = self.df2SourceCat(df[inBBox], exposure)
         return refCat
 
-    def df2SourceCat(self, df):
+    def df2SourceCat(self, df, exposure):
```

Comment:
We are getting a pixel value at a point in L834. So the user anyway has to pass in an exposure. Is there a need to keep PSF and WCS independent of the exposure?

Reply:
Since `exposure` is new here, I don't care as much as in the other signature, but making them the same in this respect - taking `exposure` and an optional `wcs` - would probably be best.